code/modules/reagents/chemistry/reagents/alcohol_reagents.dm

```diff
@@ -501,6 +510,24 @@ All effects don't start immediately, but rather get worse over time; the rate is
 	glass_icon_state = "tequilasunriseglass"
 	glass_name = "tequila Sunrise"
 	glass_desc = "Oh great, now you feel nostalgic about sunrises back on Terra..."
+	var/obj/effect/light_holder
+
+/datum/reagent/consumable/ethanol/tequila_sunrise/on_mob_add(mob/living/M)
```

Comment:
Willing to merge the rest of these but dont think we need a drink that serves no purpose except passive immunity to an antagonist

Reply:
I'd still like to have this functionality, although I'll remove it entirely if that's what you want. Would it be acceptable make the reagent delete itself if it loses the light holder instead, rather than regenerating it constantly?

Reply:
Sure that works

Reply:
Done